### ISSUE TITLE
Check H5py dependency between Conda and Mantid

### ISSUE THREAD

**Issue by glass-ships** [neutrons/drtsans#721]
Description:
============
Start reproducing the h5py importing error with following steps on the analysis cluster:

* enable dev tools with `scl enable devtoolset-7 bash`
* deactivate local _miniconda_ with `conda deactivate`
  > this will ensure Mantid is built against system library as recommended, see [here](http://developer.mantidproject.org/GettingStarted.html) for more details
* build Mantid as usual
  > this will use system Python3.6 and corresponding packages to build Mantid Python interface, which can be confirmed when invoking `cmake3`
* check out the `next` branch of `sans_backend`
* create new env with `conda create -n h5py_troubleshooting python=3.6`
* activate the new env with `conda activate h5py_troubleshooting`
* install packages with `pip install -r requirements.txt -r requirements_dev.txt`
  * here you will encounter the first error (see below), a quick workaround would be install two files separately.
```bash
ERROR: Double requirement given: pytest>=3.2 (from -r requirements_dev.txt (line 3)) (already in pytest (from -r requirements.txt (line 1)), name='pytest')
```
  > - more details about setting up sans_backend development env can be found in https://code.ornl.gov/sns-hfir-scse/sans/sans-backend  
  > - this will by default install `h5py_2.10.0`, which is much newer than the system one.
* do a dirty install of `sans_backend` with `pip install -e .`
* fire up the Ipython terminal and issue the following command 
```python
from drtsans.mono.biosans import cg3_spice_to_nexus
```
and you will get the following error
```python
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-1-07e285f20c69> in <module>
----> 1 from drtsans.mono.biosans import cg3_spice_to_nexus

~/Github/sans-backend/drtsans/__init__.py in <module>
      7 # Ordered alphabetically
      8 #######
----> 9 from .api import *
     10 from .auto_wedge import *
     11 from .beam_finder import *

~/Github/sans-backend/drtsans/api.py in <module>
----> 1 from drtsans.dataobjects import getDataType, DataType, IQazimuthal
      2 from drtsans.settings import unique_workspace_dundername as uwd
      3 # https://docs.mantidproject.org/nightly/algorithms/CloneWorkspace-v1.html
      4 # https://docs.mantidproject.org/nightly/algorithms/CreateSingleValuedWorkspace-v1.html
      5 # https://docs.mantidproject.org/nightly/algorithms/Minus-v1.html

~/Github/sans-backend/drtsans/dataobjects.py in <module>
      5 
      6 # https://docs.mantidproject.org/nightly/algorithms/CreateWorkspace-v1.html
----> 7 from mantid.simpleapi import mtd, CreateWorkspace
      8 
      9 from drtsans.settings import unique_workspace_dundername as uwd

ModuleNotFoundError: No module named 'mantid'
```
this is because Mantid is built with system Python, and it is not in your `PYTHONPATH`.
* put Mandit lib in conda Python path by
```bash
export PYTHONPATH=$MANTID_INSTALL_DIR/bin:$MANTID_INSTALL_DIR/lib:$MANTID_INSTALL_DIR/plugins
``` 
* Now try again with the importing in Ipython
```python
from drtsans.mono.biosans import cg3_spice_to_nexus
```
and you will get a different error
```python
In [1]: from drtsans.mono.biosans import cg3_spice_to_nexus
FrameworkManager-[Notice] Welcome to Mantid 5.1.0
FrameworkManager-[Notice] Please cite: http://dx.doi.org/10.1016/j.nima.2014.07.029 and this release: http://dx.doi.org/10.5286/Software/Mantid5.1
DownloadInstrument-[Notice] All instrument definitions up to date
/SNS/users/8cz/opt/miniconda3/envs/h5py_troubleshooting/lib/python3.6/site-packages/h5py/__init__.py:40: UserWarning: h5py is running against HDF5 1.8.12 when it was built against 1.10.4, this may cause problems
  '{0}.{1}.{2}'.format(*version.hdf5_built_version_tuple)
Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.10.4, library is 1.8.12
	    SUMMARY OF THE HDF5 CONFIGURATION
	    =================================

General Information:
-------------------
		   HDF5 Version: 1.8.12
		  Configured on: Wed May  8 18:40:19 UTC 2019
		  Configured by: mockbuild@buildvm-03.phx2.fedoraproject.org
		 Configure mode: production
		    Host system: x86_64-redhat-linux-gnu
	      Uname information: Linux buildvm-03.phx2.fedoraproject.org 5.0.6-200.fc29.x86_64 #1 SMP Wed Apr 3 15:09:51 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
		       Byte sex: little-endian
		      Libraries: static, shared
	     Installation point: /usr

Compiling Options:
------------------
               Compilation Mode: production
                     C Compiler: /usr/bin/gcc ( gcc (GCC) 4.8.5 20150623 )
                         CFLAGS: -O0 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic
                      H5_CFLAGS: -std=c99 -pedantic -Wall -Wextra -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wnested-externs -Winline -Wfloat-equal -Wmissing-format-attribute -Wmissing-noreturn -Wpacked -Wdisabled-optimization -Wformat=2 -Wunreachable-code -Wendif-labels -Wdeclaration-after-statement -Wold-style-definition -Winvalid-pch -Wvariadic-macros -Winit-self -Wmissing-include-dirs -Wswitch-default -Wswitch-enum -Wunused-macros -Wunsafe-loop-optimizations -Wc++-compat -Wstrict-overflow -Wlogical-op -Wlarger-than=2048 -Wvla -Wsync-nand -Wframe-larger-than=16384 -Wpacked-bitfield-compat -Wstrict-overflow=5 -Wjump-misses-init -Wunsuffixed-float-constants -Wdouble-promotion -Wsuggest-attribute=const -Wtrampolines -Wstack-usage=8192 -Wvector-operation-performance -Wsuggest-attribute=pure -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -O3 -fomit-frame-pointer -finline-functions
                      AM_CFLAGS: 
                       CPPFLAGS: 
                    H5_CPPFLAGS: -D_POSIX_C_SOURCE=199506L   -DNDEBUG -UH5_DEBUG_API
                    AM_CPPFLAGS: -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_BSD_SOURCE 
               Shared C Library: yes
               Static C Library: yes
  Statically Linked Executables: no
                        LDFLAGS: -Wl,-z,relro 
                     H5_LDFLAGS: 
                     AM_LDFLAGS: 
 	 	Extra libraries:  -lsz -lz -ldl -lm 
 		       Archiver: ar
 		 	 Ranlib: ranlib
 	      Debugged Packages: 
		    API Tracing: no

Languages:
----------
                        Fortran: yes
               Fortran Compiler: /usr/bin/gfortran ( GNU Fortran (GCC) 4.8.5 20150623 )
          Fortran 2003 Compiler: yes
                  Fortran Flags: -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic -I/usr/lib64/gfortran/modules
               H5 Fortran Flags:  
               AM Fortran Flags: 
         Shared Fortran Library: yes
         Static Fortran Library: yes

                            C++: yes
                   C++ Compiler: /usr/bin/g++ ( g++ (GCC) 4.8.5 20150623 )
                      C++ Flags: -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic
                   H5 C++ Flags:  
                   AM C++ Flags: 
             Shared C++ Library: yes
             Static C++ Library: yes

Features:
---------
                  Parallel HDF5: no
             High Level library: yes
                   Threadsafety: no
            Default API Mapping: v18
 With Deprecated Public Symbols: yes
         I/O filters (external): deflate(zlib),szip(encoder)
         I/O filters (internal): shuffle,fletcher32,nbit,scaleoffset
                            MPE: no
                     Direct VFD: no
                        dmalloc: no
Clear file buffers before write: yes
           Using memory checker: no
         Function Stack Tracing: no
                           GPFS: no
      Strict File Format Checks: no
   Optimization Instrumentation: no
       Large File Support (LFS): yes
Bye...
Aborted (core dumped)
```
TL;DR, HDF5 detected that two different versions of h5py are used
  * 2.7.1 for `Mantid` from system
  * 2.10.0 for `sans_backend` from conda   
and forcefully stopping us from moving forward.
We could bypass the version check by setting `HDF5_DISABLE_VERSION_CHECK` to 1 as suggested in the error messages.
* Bypassing HDF5 version check with
```bash
export HDF5_DISABLE_VERSION_CHECK=1
```
and repeat the previous step, now we are getting `segmentation fault` instead of `aborting`
```python
Python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.16.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from drtsans.mono.biosans import cg3_spice_to_nexus
FrameworkManager-[Notice] Welcome to Mantid 5.1.0
FrameworkManager-[Notice] Please cite: http://dx.doi.org/10.1016/j.nima.2014.07.029 and this release: http://dx.doi.org/10.5286/Software/Mantid5.1
DownloadInstrument-[Notice] All instrument definitions up to date
/SNS/users/8cz/opt/miniconda3/envs/h5py_troubleshooting/lib/python3.6/site-packages/h5py/__init__.py:40: UserWarning: h5py is running against HDF5 1.8.12 when it was built against 1.10.4, this may cause problems
  '{0}.{1}.{2}'.format(*version.hdf5_built_version_tuple)
Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
'HDF5_DISABLE_VERSION_CHECK' environment variable is set to 1, application will
continue at your own risk.
Headers are 1.10.4, library is 1.8.12
	    SUMMARY OF THE HDF5 CONFIGURATION
	    =================================

General Information:
-------------------
		   HDF5 Version: 1.8.12
		  Configured on: Wed May  8 18:40:19 UTC 2019
		  Configured by: mockbuild@buildvm-03.phx2.fedoraproject.org
		 Configure mode: production
		    Host system: x86_64-redhat-linux-gnu
	      Uname information: Linux buildvm-03.phx2.fedoraproject.org 5.0.6-200.fc29.x86_64 #1 SMP Wed Apr 3 15:09:51 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
		       Byte sex: little-endian
		      Libraries: static, shared
	     Installation point: /usr

Compiling Options:
------------------
               Compilation Mode: production
                     C Compiler: /usr/bin/gcc ( gcc (GCC) 4.8.5 20150623 )
                         CFLAGS: -O0 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic
                      H5_CFLAGS: -std=c99 -pedantic -Wall -Wextra -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wnested-externs -Winline -Wfloat-equal -Wmissing-format-attribute -Wmissing-noreturn -Wpacked -Wdisabled-optimization -Wformat=2 -Wunreachable-code -Wendif-labels -Wdeclaration-after-statement -Wold-style-definition -Winvalid-pch -Wvariadic-macros -Winit-self -Wmissing-include-dirs -Wswitch-default -Wswitch-enum -Wunused-macros -Wunsafe-loop-optimizations -Wc++-compat -Wstrict-overflow -Wlogical-op -Wlarger-than=2048 -Wvla -Wsync-nand -Wframe-larger-than=16384 -Wpacked-bitfield-compat -Wstrict-overflow=5 -Wjump-misses-init -Wunsuffixed-float-constants -Wdouble-promotion -Wsuggest-attribute=const -Wtrampolines -Wstack-usage=8192 -Wvector-operation-performance -Wsuggest-attribute=pure -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -O3 -fomit-frame-pointer -finline-functions
                      AM_CFLAGS: 
                       CPPFLAGS: 
                    H5_CPPFLAGS: -D_POSIX_C_SOURCE=199506L   -DNDEBUG -UH5_DEBUG_API
                    AM_CPPFLAGS: -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_BSD_SOURCE 
               Shared C Library: yes
               Static C Library: yes
  Statically Linked Executables: no
                        LDFLAGS: -Wl,-z,relro 
                     H5_LDFLAGS: 
                     AM_LDFLAGS: 
 	 	Extra libraries:  -lsz -lz -ldl -lm 
 		       Archiver: ar
 		 	 Ranlib: ranlib
 	      Debugged Packages: 
		    API Tracing: no

Languages:
----------
                        Fortran: yes
               Fortran Compiler: /usr/bin/gfortran ( GNU Fortran (GCC) 4.8.5 20150623 )
          Fortran 2003 Compiler: yes
                  Fortran Flags: -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic -I/usr/lib64/gfortran/modules
               H5 Fortran Flags:  
               AM Fortran Flags: 
         Shared Fortran Library: yes
         Static Fortran Library: yes

                            C++: yes
                   C++ Compiler: /usr/bin/g++ ( g++ (GCC) 4.8.5 20150623 )
                      C++ Flags: -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic
                   H5 C++ Flags:  
                   AM C++ Flags: 
             Shared C++ Library: yes
             Static C++ Library: yes

Features:
---------
                  Parallel HDF5: no
             High Level library: yes
                   Threadsafety: no
            Default API Mapping: v18
 With Deprecated Public Symbols: yes
         I/O filters (external): deflate(zlib),szip(encoder)
         I/O filters (internal): shuffle,fletcher32,nbit,scaleoffset
                            MPE: no
                     Direct VFD: no
                        dmalloc: no
Clear file buffers before write: yes
           Using memory checker: no
         Function Stack Tracing: no
                           GPFS: no
      Strict File Format Checks: no
   Optimization Instrumentation: no
       Large File Support (LFS): yes
Segmentation fault (core dumped)
```
* A temp workaround here would be uninstall the newer `h5py`, then symbolic link the system `h5py` to the `site-package` directory of the environment, which will resolve the error seeing above.

List of Tasks:
---------------
* [x]  Understand Story: Talk to owner
* [x]  Address Necessary Revisions to the Story
       
* [ ]  Capture Significant Design Decisions
* [ ]  Gather All Required Test Data
       
       
* [ ]  Write & Run Unit Tests
* [ ]  Implement Story (Update unit test as needed)
* [ ]  Write & Run Integration Tests
* [ ]  Submit Pull Request
* [ ]  Report Completion During Status Meeting

**Comment by glass-ships** [comment]
This issue was migrated from [code.ornl.gov](https://code.ornl.gov/sns-hfir-scse/sans/sans-backend/-/issues/676).
